app/src/main/java/com/yml/womensafety/navigationdrawer/home/HomePageActivity.kt

```diff
@@ -57,48 +61,54 @@ class HomePageActivity : AppCompatActivity() {
         startAccelerometerService()
         receiveBroadcast()
 
-        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottomNavBar)
-        bottomNavigationView.apply {
-            background = null
-            menu.getItem(INDEX).isEnabled = false
-            itemIconTintList = null
+        bottomNavigation.apply {
+            show(ID_HOME)
+            add(MeowBottomNavigation.Model(ID_HOME, R.drawable.ic_outline_home))
+            add(MeowBottomNavigation.Model(ID_CONTACTS, R.drawable.ic_add_contacts))
+            add(MeowBottomNavigation.Model(ID_PROFILE, R.drawable.ic_outline_account))
+            add(MeowBottomNavigation.Model(ID_LOGOUT, R.drawable.ic_exit_to_app))
         }
         contactsList = mutableListOf()
 
         contactsViewModel = ViewModelProvider(this).get(ContactsViewModel()::class.java)
         contactsViewModel.initializeRepository()
 
-        bottomNavigationView.setOnNavigationItemReselectedListener {
-            when (it.itemId) {
-                R.id.navHome -> {
-                    startActivity(Intent(this, HomePageActivity::class.java))
-                }
-                R.id.navLogout -> {
-                    FirebaseUtil.user?.signOut()
-                    startActivity(Intent(this, LoginActivity::class.java))
-                    finish()
+
+        bottomNavigation.setOnClickMenuListener {
+            when (it.id) {
+                ID_HOME -> {
+                    homepageFragment = HomePageFragment()
+                    FragmentUtil.replaceFragment(
+                        supportFragmentManager,
+                        homepageFragment,
+                        R.id.myNavHostFragment
+                    )
                 }
-                R.id.navContacts -> {
+                ID_CONTACTS -> {
                     contactsFragment = ContactsFragment()
-                    FragmentUtil.addFragmentToActivity(
+                    FragmentUtil.replaceFragment(
                         supportFragmentManager,
                         contactsFragment,
                         R.id.myNavHostFragment
                     )
                 }
-                R.id.navProfile -> {
+
+                ID_PROFILE -> {
                     userProfile = UserProfile()
```

Comment:
Why do we require a global reference of all the fragments?

Reply:
done.